src/Editor/main.cpp

```diff
@@ -62,53 +52,65 @@ int main() {
         float deltaTime = static_cast<float>(glfwGetTime() - lastTime);
         lastTime = glfwGetTime();
         
+        Managers().profilingManager->SetActive(profiling);
+
         // Begin new profiling frame.
-        Managers().profilingManager->BeginFrame();
+        if (Managers().profilingManager->Active())
+            Managers().profilingManager->BeginFrame();
         
         { PROFILE("Frame");
+        { GPUPROFILE("Frame", Video::Query::Type::TIME_ELAPSED);
+
             glfwPollEvents();
-            
+
             if (Input()->Triggered(InputHandler::PROFILE))
                 profiling = !profiling;
-            
+
             // Start new frame.
             ImGuiImplementation::NewFrame();
-            
+
             window->Update();
-            
+
             if (editor->IsVisible()) {
                 Hymn().world.ClearKilled();
                 Managers().particleManager->Update(Hymn().world, deltaTime, true);
+
                 Managers().debugDrawingManager->Update(deltaTime);
                 Hymn().Render(editor->GetCamera(), EditorSettings::GetInstance().GetBool("Sound Source Icons"), EditorSettings::GetInstance().GetBool("Particle Emitter Icons"), EditorSettings::GetInstance().GetBool("Light Source Icons"), EditorSettings::GetInstance().GetBool("Camera Icons"), EditorSettings::GetInstance().GetBool("Physics Volumes"), EditorSettings::GetInstance().GetBool("Grid Settings"));
-                
+
                 if (window->ShouldClose())
                     editor->Close();
-    
+
                 editor->Show(deltaTime);
 
                 if (window->ShouldClose() && !editor->isClosing())
                     window->CancelClose();
 
             } else {
                 { PROFILE("Update");
+                { GPUPROFILE("Update", Video::Query::Type::TIME_ELAPSED);
```

Comment:
We may want to indent this w.r.t. the outer profile, or put it on the same line.